### PR TITLE
PS-3013 Fix stamped field ids

### PR DESCRIPTION
**Pull request by citizengabe**
## What it does

In order to avoid `id` collisions for DOM elements in sections stamped out from templates, I'm building the keys using the indices of `TemplateSequence.values`.

### CODE REVIEW

**Comment by citizengabe** [comment]
I'm leaving this in draft status until I get some UI testing in place.

File: src/Lynx.purs [hunk 1]
```diff
@@ -140,29 +140,29 @@ component =
...
+      $ Data.Array.fromFoldable $ renderField (map show index) <$> section.fields
 
   renderSequence :: Sequence ExprType -> H.ParentHTML Query (ChildQuery m) ChildSlot m
   renderSequence sequence =
     Layout.container_
       ( [ Format.subHeading_ [ HH.text sequence.name ] ]
         <> case sequence.values of
-            UserInput sections -> renderSection <$> sections
-            Invalid sections -> renderSection <$> sections
+            UserInput sections -> Data.Array.mapWithIndex (renderSection <<< Just) sections
```
Comment (citizengabe) [inline src/Lynx.purs]: Should I be using `FunctorWithIndex` here?

Comment (joneshf-cn) [reply src/Lynx.purs]: I don't think it much matters. They ought to be equivalent if not the same.

Comment (citizengabe) [reply src/Lynx.purs]: I ended up changing it anyway

**Review by joneshf-cn — APPROVED** [review]
Looks good!

**Pull request merged**